website/blog/testing.md

```diff
@@ -0,0 +1,512 @@
+            Address: shippingservice.Address{
+                StreetAddress: "1600 Amphitheatre Parkway",
+                City:          "Mountain View",
+                State:         "CA",
+                Country:       "USA",
+                ZipCode:       94043,
+            },
+            Email: "sundar@google.com",
+            CreditCard: paymentservice.CreditCardInfo{
+                Number:          "4432-8015-6152-0454",
+                CVV:             672,
+                ExpirationYear:  2025,
+                ExpirationMonth: time.January,
+            },
+        }
+        if _, err := checkout.PlaceOrder(ctx, order); err != nil {
+            t.Fatal(err)
+        }
+    })
+}
+```
+
+The integration test lists all available products using the catalog service,
+adds one of every product to a cart using the cart service, and checks out using
+the checkout service. Service Weaver automatically initializes all these
+services, as well as the other services on which they depend (e.g., shipping
+service, email service, payment service, etc.). Moreover, Service Weaver reduces
+the tedium of writing integration tests by allowing you to use language native
+types and method calls to interact with various services.
+
+We can run this integration test as easily as running `go test`:
+
+```console
+$ go test
+PASS
+ok      github.com/ServiceWeaver/onlineboutique 0.026s
+```
+
+In addition to using the `weavertest.Local` runner, which runs all components
+locally in a single process, we can also use
+
+- the `weavertest.RPC` runner, which forces components to interact via RPC, and
+- the `weavertest.Multi` runner, which runs every component is a separate
+  OS process.
+
+## 3. Randomized Testing
+
+When you write a unit test or integration test, you write some code and check
+that it executes in the way you expect. For example, to test a `Reversed([]byte)
+[]byte` function, you might check that
+
+- `Reversed([]byte{})` is `[]byte{}`,
+- `Reversed([]byte{1})` is `[]byte{1}`,
+- `Reversed([]byte{1, 2})` is `[]byte{2, 1}`, and so on.
+
+Unit and integration tests are thus limited to only test the behaviors and
+corner cases that you can think of and have the patience to write down.
+
+With randomized **property-based tests**, on the other hand, you specify a
+property of your code that should always hold and then test the property on
+millions of randomly generated examples. For example, we can test that the
+`Reversed` function is [involutive][]&mdash;i.e. that `Reversed(Reversed(x)) ==
+x`&mdash;by writing a [fuzz test][fuzzing] that tests the property on randomly
+generated byte slices:
+
+```go
+func FuzzReversedIsInvolutive(f *testing.F) {
+    f.Fuzz(func(t *testing.T, want []byte) {
+        got := Reversed(Reversed(want))
+        if !slices.Equal(want, got) {
+            t.Fatalf("got %v, want %v", got, want)
+        }
+    })
+}
+```
+
+Service Weaver allows you to apply randomized property-based testing to entire
```

Comment:
Can we say something on how much more challenging is to write these kind of fuzzing tests w/o Service Weaver?